kedro/io/partitioned_data_set.py

```diff
@@ -58,6 +59,18 @@
 S3_PROTOCOLS = ("s3", "s3a", "s3n")
 
 
+def _grandparent(path: str) -> str:
+    path_obj = PurePosixPath(path)
+    grandparent = path_obj.parents[1]
+    if grandparent.name != path_obj.name:
+        last_three_parts = path_obj.relative_to(*path_obj.parts[:-3])
+        raise DataSetError(
+            f"`{path}` is not a well-formed versioned path ending with "
+            f"`filename/timestamp/filename` (got `{last_three_parts}`)."
+        )
+    return str(grandparent)
+
+
 class PartitionedDataSet(AbstractDataSet):
```

Comment:
Hey @deepyaman! I believe if we want this to be merged before 0.17, we might need these changes to be implemented for the `IncrementalDataSet` too 🤔  (since `_list_partitions` method for the incremental one isn't inherited from the `PartitionedDataSet`). Please let me know whether it makes sense? 🙂

Reply:
Also should this merged in `develop` instead?